> > Back to the original issue, removing duplicates from the function arguments list, instead of `@options(...)` list, is the right way to solve this issue.
> 
> I fully agree. Having some of the arguments of `@options` in the function argument list is not correct. They should be removed from the function argument list, not from `@options`.

I tried this, it just doesn't work. One has to understand how that blasted `@option` work better than me.
I'm getting
```
sage -t --warn-long 21.9 --random-seed=329640674077002804385470819635015891914 src/sage/plot/contour_plot.py
**********************************************************************
File "src/sage/plot/contour_plot.py", line 1203, in sage.plot.contour_plot.implicit_plot
Failed example:
    implicit_plot(x**2 + y**2 == 2, (x,-3,3), (y,-3,3), fill=True, fillcolor='red')
Expected:
    Graphics object consisting of 2 graphics primitives
Got:
    verbose 0 (163: primitive.py, options) WARNING: Ignoring option 'incol'=red
    verbose 0 (163: primitive.py, options) WARNING: Ignoring option 'outcol'=None
    verbose 0 (163: primitive.py, options) WARNING: Ignoring option 'bordercol'=blue
    verbose 0 (163: primitive.py, options) WARNING: Ignoring option 'borderstyle'=None
    verbose 0 (163: primitive.py, options) WARNING: Ignoring option 'borderwidth'=None
    verbose 0 (163: primitive.py, options) WARNING: Ignoring option 'alpha'=1
    verbose 0 (163: primitive.py, options) 
    The allowed options for ContourPlot defined by a 150 x 150 data grid are:
        cmap           the name of a predefined colormap,
                            a list of colors, or an instance of a
                            matplotlib Colormap. Type: import matplotlib.cm;
                            matplotlib.cm.datad.keys()
                            for available colormap names.
        colorbar       Include a colorbar indicating the levels                    
        colorbar_optionsa dictionary of options for colorbars                       
        contours       Either an integer specifying the number of
                            contour levels, or a sequence of numbers giving
                            the actual contours to use.
        fill           Fill contours or not                                        
        label_options  a dictionary of options for the labels                      
        labels         show line labels or not                                     
        legend_label   The label for this item in the legend.                      
        linestyles     the style of the lines to be plotted                        
        linewidths     the width of the lines to be plotted                        
        plot_points    How many points to use for plotting precision               
        zorder         The layer level in which to draw                            
    <BLANKLINE>
    verbose 0 (163: primitive.py, options) WARNING: Ignoring option 'incol'=red
    verbose 0 (163: primitive.py, options) WARNING: Ignoring option 'outcol'=None
    verbose 0 (163: primitive.py, options) WARNING: Ignoring option 'bordercol'=blue
    verbose 0 (163: primitive.py, options) WARNING: Ignoring option 'borderstyle'=None
    verbose 0 (163: primitive.py, options) WARNING: Ignoring option 'borderwidth'=None
    verbose 0 (163: primitive.py, options) WARNING: Ignoring option 'alpha'=1
    verbose 0 (163: primitive.py, options) 
    The allowed options for ContourPlot defined by a 150 x 150 data grid are:
        cmap           the name of a predefined colormap,
                            a list of colors, or an instance of a
                            matplotlib Colormap. Type: import matplotlib.cm;
                            matplotlib.cm.datad.keys()
                            for available colormap names.
        colorbar       Include a colorbar indicating the levels                    
        colorbar_optionsa dictionary of options for colorbars                       
        contours       Either an integer specifying the number of
                            contour levels, or a sequence of numbers giving
                            the actual contours to use.
        fill           Fill contours or not                                        
        label_options  a dictionary of options for the labels                      
        labels         show line labels or not                                     
        legend_label   The label for this item in the legend.                      
        linestyles     the style of the lines to be plotted                        
        linewidths     the width of the lines to be plotted                        
        plot_points    How many points to use for plotting precision               
        zorder         The layer level in which to draw                            
    <BLANKLINE>
    verbose 0 (163: primitive.py, options) WARNING: Ignoring option 'incol'=red
    verbose 0 (163: primitive.py, options) WARNING: Ignoring option 'outcol'=None
    verbose 0 (163: primitive.py, options) WARNING: Ignoring option 'bordercol'=blue
    verbose 0 (163: primitive.py, options) WARNING: Ignoring option 'borderstyle'=None
    verbose 0 (163: primitive.py, options) WARNING: Ignoring option 'borderwidth'=None
    verbose 0 (163: primitive.py, options) WARNING: Ignoring option 'alpha'=1
    verbose 0 (163: primitive.py, options) 
    The allowed options for ContourPlot defined by a 150 x 150 data grid are:
        cmap           the name of a predefined colormap,
                            a list of colors, or an instance of a
                            matplotlib Colormap. Type: import matplotlib.cm;
                            matplotlib.cm.datad.keys()
                            for available colormap names.
        colorbar       Include a colorbar indicating the levels                    
        colorbar_optionsa dictionary of options for colorbars                       
        contours       Either an integer specifying the number of
                            contour levels, or a sequence of numbers giving
                            the actual contours to use.
        fill           Fill contours or not                                        
        label_options  a dictionary of options for the labels                      
        labels         show line labels or not                                     
        legend_label   The label for this item in the legend.                      
        linestyles     the style of the lines to be plotted                        
        linewidths     the width of the lines to be plotted                        
        plot_points    How many points to use for plotting precision               
        zorder         The layer level in which to draw                            
    <BLANKLINE>
    verbose 0 (163: primitive.py, options) WARNING: Ignoring option 'incol'=red
    verbose 0 (163: primitive.py, options) WARNING: Ignoring option 'outcol'=None
    verbose 0 (163: primitive.py, options) WARNING: Ignoring option 'bordercol'=blue
    verbose 0 (163: primitive.py, options) WARNING: Ignoring option 'borderstyle'=None
    verbose 0 (163: primitive.py, options) WARNING: Ignoring option 'borderwidth'=None
    verbose 0 (163: primitive.py, options) WARNING: Ignoring option 'alpha'=1
    verbose 0 (163: primitive.py, options) 
    The allowed options for ContourPlot defined by a 150 x 150 data grid are:
        cmap           the name of a predefined colormap,
                            a list of colors, or an instance of a
                            matplotlib Colormap. Type: import matplotlib.cm;
                            matplotlib.cm.datad.keys()
                            for available colormap names.
        colorbar       Include a colorbar indicating the levels                    
        colorbar_optionsa dictionary of options for colorbars                       
        contours       Either an integer specifying the number of
                            contour levels, or a sequence of numbers giving
                            the actual contours to use.
        fill           Fill contours or not                                        
...
```
incomprehensible errors if I do
```diff
+++ b/src/sage/plot/contour_plot.py
@@ -1392,10 +1392,9 @@ def implicit_plot(f, xrange, yrange, **options):
 
 
 @options(plot_points=100, incol='blue', outcol=None, bordercol=None,
-         borderstyle=None, borderwidth=None, frame=False, axes=True,
-         legend_label=None, aspect_ratio=1, alpha=1)
-def region_plot(f, xrange, yrange, plot_points, incol, outcol, bordercol,
-                borderstyle, borderwidth, alpha, **options):
+         borderstyle=None, borderwidth=None, alpha=1,
+         frame=False, axes=True, legend_label=None, aspect_ratio=1)
+def region_plot(f, xrange, yrange, **options):
     r"""
     ``region_plot`` takes a boolean function of two variables, `f(x, y)`
     and plots the region where f is True over the specified
@@ -1659,6 +1658,14 @@ def region_plot(f, xrange, yrange, plot_points, incol, outcol, bordercol,
     from warnings import warn
     import numpy
 
+    plot_points = options['plot_points']
+    incol = options['incol']
+    outcol = options['outcol']
+    bordercol = options['bordercol']
+    borderstyle = options['borderstyle']
+    borderwidth = options['borderwidth']
+    alpha = options['alpha']
+
     if not isinstance(f, (list, tuple)):
         f = [f]
 
...
*********************************************************************
File "src/sage/plot/contour_plot.py", line 1650, in sage.plot.contour_plot.region_plot
Failed example:
    region_plot([x == 0], (x,-1,1), (y,-1,1))
Exception raised:
    Traceback (most recent call last):
      File "/home/scratch/scratch2/dimpase/sage/sage/src/sage/doctest/forker.py", line 709, in _run
        self.compile_and_execute(example, compiler, test.globs)
      File "/home/scratch/scratch2/dimpase/sage/sage/src/sage/doctest/forker.py", line 1144, in compile_and_execute
        exec(compiled, globs)
      File "<doctest sage.plot.contour_plot.region_plot[21]>", line 1, in <module>
        region_plot([x == Integer(0)], (x,-Integer(1),Integer(1)), (y,-Integer(1),Integer(1)))
      File "/home/scratch/scratch2/dimpase/sage/sage/src/sage/misc/decorators.py", line 497, in wrapper
        return func(*args, **options)
               ^^^^^^^^^^^^^^^^^^^^^^
      File "/home/scratch/scratch2/dimpase/sage/sage/src/sage/plot/contour_plot.py", line 1687, in region_plot
        return implicit_plot(feqs[0], xrange, yrange, plot_points=plot_points,
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    TypeError: sage.plot.contour_plot.implicit_plot() got multiple values for keyword argument 'plot_points'
...
```

and if I leave out assignments 
```
    plot_points = options['plot_points']
    incol = options['incol']
...
```
in the function body I get errors about undefined local variables:
```
**********************************************************************
File "src/sage/plot/contour_plot.py", line 1203, in sage.plot.contour_plot.implicit_plot
Failed example:
    implicit_plot(x**2 + y**2 == 2, (x,-3,3), (y,-3,3), fill=True, fillcolor='red')
Exception raised:
    Traceback (most recent call last):
      File "/home/scratch/scratch2/dimpase/sage/sage/src/sage/doctest/forker.py", line 709, in _run
        self.compile_and_execute(example, compiler, test.globs)
      File "/home/scratch/scratch2/dimpase/sage/sage/src/sage/doctest/forker.py", line 1144, in compile_and_execute
        exec(compiled, globs)
      File "<doctest sage.plot.contour_plot.implicit_plot[8]>", line 1, in <module>
        implicit_plot(x**Integer(2) + y**Integer(2) == Integer(2), (x,-Integer(3),Integer(3)), (y,-Integer(3),Integer(3)), fill=True, fillcolor='red')
      File "/home/scratch/scratch2/dimpase/sage/sage/src/sage/misc/decorators.py", line 497, in wrapper
        return func(*args, **options)
               ^^^^^^^^^^^^^^^^^^^^^^
      File "/home/scratch/scratch2/dimpase/sage/sage/src/sage/plot/contour_plot.py", line 1382, in implicit_plot
        return region_plot(f < 0, xrange, yrange, borderwidth=linewidths,
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/home/scratch/scratch2/dimpase/sage/sage/src/sage/misc/decorators.py", line 497, in wrapper
        return func(*args, **options)
               ^^^^^^^^^^^^^^^^^^^^^^
      File "/home/scratch/scratch2/dimpase/sage/sage/src/sage/plot/contour_plot.py", line 1692, in region_plot
        plot_points)
        ^^^^^^^^^^^
    NameError: name 'plot_points' is not defined
```

Unless we dig up someone who knows about `@options`, this is not going to fly. Or perhaps it's just not fixable this way, full stop.

--- a/src/sage/plot/contour_plot.py
+++ b/src/sage/plot/contour_plot.py
@@ -1394,8 +1394,7 @@ def implicit_plot(f, xrange, yrange, **options):
 @options(plot_points=100, incol='blue', outcol=None, bordercol=None,
          borderstyle=None, borderwidth=None, frame=False, axes=True,
          legend_label=None, aspect_ratio=1, alpha=1)
-def region_plot(f, xrange, yrange, plot_points, incol, outcol, bordercol,
-                borderstyle, borderwidth, alpha, **options):
+def region_plot(f, xrange, yrange, **options):
     r"""
     ``region_plot`` takes a boolean function of two variables, `f(x, y)`
     and plots the region where f is True over the specified
@@ -1659,6 +1658,14 @@ def region_plot(f, xrange, yrange, plot_points, incol, outcol, bordercol,
     from warnings import warn
     import numpy
 
+    plot_points = options['plot_points']
+    incol = options.pop('incol')
+    outcol = options.pop('outcol')
+    bordercol = options.pop('bordercol')
+    borderstyle = options.pop('borderstyle')
+    borderwidth = options.pop('borderwidth')
+    alpha = options.pop('alpha')
+
     if not isinstance(f, (list, tuple)):
         f = [f]
 
@@ -1677,9 +1684,9 @@ def region_plot(f, xrange, yrange, plot_points, incol, outcol, bordercol,
     if neqs and not bordercol:
         bordercol = incol
     if not f:
-        return implicit_plot(feqs[0], xrange, yrange, plot_points=plot_points,
-                             fill=False, linewidth=borderwidth,
-                             linestyle=borderstyle, color=bordercol, **options)
+        return implicit_plot(feqs[0], xrange, yrange, fill=False,
+                             linewidth=borderwidth, linestyle=borderstyle,
+                             color=bordercol, **options)
     f_all, ranges = setup_for_eval_on_grid(feqs + f,
                                            [xrange, yrange],
                                            plot_points)

